File: Assignment_20/program1.c
///////////////////////////////////////////////////////////////////////////////
// File Name   :  program3.c 
// Author      :  Sanket Ashok Katrajkar
// Date        :  04/06/2025
///////////////////////////////////////////////////////////////////////////////
//
// Description : Accept N numbers from user and one number NO,
//               and return the NO is present or not
//
/////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdbool.h>
#include<stdlib.h>

#define TRUE 1
#define FALSE 0

typedef int BOOL;

int check(int Arr[], int iLength, int iNo)
{
    int iCnt = 0, iCount = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        if(Arr[iCnt] == iNo)
        {
            return true;
        }
    }
    return false;
}

int main()
{
    int iLength = 0, iCnt = 0, iValue = 0;
    int *ptr = NULL;
    BOOL  iRet = false;

    printf("Enter the number of elements:\n");
    scanf("%d", &iLength);

    printf("Enter the number to check frequency:\n");
    scanf("%d", &iValue);

    ptr = (int *)malloc(iLength * sizeof(int));
    if(ptr == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter the elements:\n");
    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        printf("Enter element %d: ", iCnt + 1);
        scanf("%d", &ptr[iCnt]);
    }

    iRet = check(ptr, iLength, iValue);

    if(iRet == true )
    {
        printf("Number  is present");
    }
    else
    {
        printf("Number is  not present");
    }

    free(ptr);

    return 0;
}

File: Assingmentd_44/program1.java
///////////////////////////////////////////////////////////////////////////////
// File Name   :  program1 - 5.java
// Author      :  Sanket Ashok Katrajkar
// Date        :  01/07/2025
///////////////////////////////////////////////////////////////////////////////
// 
// Description :  Accept the Matrix from user and display the additon of 
//                diagonal element.
//              
//                Accept the Matrix from user and display the additon of 
//                diagonal element.
//                
//                Accept the Matrix from user and display the  Largest element of 
//                All matrix
// 
/////////////////////////////////////////////////////////////////////////////////

/*
  Input :

    3  2  5  4
    4  3  2  2
    8  4  1  5
    3  3  7  5
 Q1
 Output: 12 

 Q2
 Input : 5
 Ouptut : 2

Q3
Output : 7

*/

import java.util.*;

class Matrix 
{
    private int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int [A][B];
    }
    
    public void Accept(int A, int B)
    {
        System.out.println("Enter the element of Matrix : ");
        
        Scanner sobj = new Scanner(System.in);
        int i , j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public void Display()
    {
        
        int i , j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
               System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }


    public int DiagonalSum()
    {
       int i , j = 0, iSum = 0 ;

       for(i = 0; i < Arr.length; i++)
       {
         for(j = 0; j < Arr[i].length; j++)
         {
            if(i == j)
            {
                iSum += Arr[i][j];
            }
         }
       }
       return iSum;
    }

    public int FrequencyCount(int iNo)
    {
       int i , j = 0, iCount = 0 ;

       for(i = 0; i < Arr.length; i++)
       {
         for(j = 0; j < Arr[i].length; j++)
         {
            if(Arr[i][j] == iNo)
            {
                iCount++;
            }
         }
       }
       return iCount;
    }

    public int LargestNumber()
    {
       int i , j = 0, iMax = 0 ;

       for(i = 0; i < Arr.length; i++)
       {
         for(j = 0; j < Arr[i].length; j++)
         {
            if(Arr[i][j] > iMax)
            {
                iMax = Arr[i][j];
            }
         }
       }
       return iMax;
    }
}

public class program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the Number of Rows");
        int iRow = sobj.nextInt();

        System.out.println("Enter the number of columns");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);
        int iRet = 0;

        mobj.Accept(iRow, iCol);

        System.out.println("Enter the number that you want to count frequency : ");
        int iNo = sobj.nextInt();

        mobj.Display(); 

        iRet = mobj.DiagonalSum(); 
        System.out.println("Addition is : "+ iRet);

        iRet = mobj.FrequencyCount(iNo); 
        System.out.println("Count is : "+ iRet);

        iRet = mobj.LargestNumber(); 
        System.out.println("Largest Number is  : "+ iRet);
    }
}




